### ISSUE TITLE
References to Objective C protocols by name broken in 1.8.12

### ISSUE THREAD

**Issue by mstorsjo** [doxygen/doxygen#6632]
Since 1.8.12 (in particular, since 00ee930a1d73e11885197102c54fd4c8141127da), direct refererences to objective C protocols fail to be resolved.

This can be observed with the following test header:

```
@protocol MyProtocol <NSObject>
/**
 * Some callback. 
 */
- (void) callbackMethod;
@end

@interface MyClass
/**
 * See #MyProtocol::callbackMethod.
 */
- (void) directMethod;
@end
```

With 1.8.11 and commits up to the referenced one, this builds fine and produces a hyperlink, rendered like this:
> See callbackMethod (MyProtocol-p).

With 1.8.12 and later, it fails to produce a hyperlink and gives a warning:
```
header.h:9: warning: explicit link request to 'MyProtocol::callbackMethod' could not be resolved
```

**Comment by mstorsjo** [comment]
This seems to be related to #3406, although the existing setup worked fine before 1.8.12.

**Comment by mstorsjo** [comment]
Thanks! As I'm unfamiliar with the doxygen internals I can't comment much about it, but I can at least confirm that it does indeed fix the full original issue I saw in addition to the reduced testcase.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.15.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).

**Comment by mstorsjo** [comment]
I can confirm that this issue indeed is fixed in 1.8.15.
